File: src/dispatcher/protocol/task.ts

export type TaskInfoV1 = {
  _v: 1,
  task_options?: TaskOptions,
  id: string,
  max_price: number,
  time_to_money_ratio: number,
}

export type TaskResultV1 = {
  _v: 1,
  image: string,
}

export type TaskInfo = TaskInfoV1;
export type TaskResult = TaskResultV1;

export type TaskOptions = {
    prompt: string,
    model: string,
    size?: string,
    steps?: number,
};

export type TaskResultClient = {
  resultsUrl: string[],
  taskId: string,
  error?: string,
  type: 'result' | 'error' | 'status',
  status?: 'ready' | 'inProgress',
};

File: backend/src/dispatcher/protocol/task.py
from dataclasses import dataclass
from typing import Optional, List
from enum import Enum

@dataclass
class TaskOptions:
    prompt: str
    model: str
    size: Optional[str] = None
    steps: Optional[int] = None

@dataclass
class TaskInfoV1:
    _v: int
    task_options: Optional[TaskOptions] = None
    id: str
    max_price: float
    time_to_money_ratio: float

@dataclass
class TaskResultV1:
  _v: int
  image: str

TaskInfo = TaskInfoV1
TaskResult = TaskResultV1

class TaskType(Enum):
    RESULT = 1
    ERROR = 2
    STATUS = 3

class TaskReadinessStatus(Enum):
    READY = 1
    IN_PROGRESS = 2

@dataclass
class TaskResultClient:
    resultsUrl: List[str]
    taskId: str
    error: Optional[str] = None
    type: TaskType
    status: Optional[TaskReadinessStatus] = None
